apps/total-typescript/src/pages/invoices/[merchantChargeId].tsx

```diff
@@ -0,0 +1,227 @@
+import * as React from 'react'
+import {DownloadIcon} from '@heroicons/react/outline'
+import {convertToSerializeForNextResponse} from '@skillrecordings/commerce-server'
+import {useLocalStorage} from 'react-use'
+import {GetServerSideProps} from 'next'
+import {stripe} from '@skillrecordings/commerce-server'
+import {getSdk} from '@skillrecordings/database'
+import {Stripe} from 'stripe'
+import fromUnixTime from 'date-fns/fromUnixTime'
+import Layout from 'components/app/layout'
+import format from 'date-fns/format'
+import {prisma} from '@skillrecordings/database'
+import {getCurrentAbility} from '@skillrecordings/ability'
+import {getToken} from 'next-auth/jwt'
+
+export const getServerSideProps: GetServerSideProps = async ({
+  res,
+  req,
+  query,
+}) => {
+  const sessionToken = await getToken({req})
+  const {merchantChargeId} = query
+  const {getProduct} = getSdk()
+  const ability = getCurrentAbility(sessionToken as any)
+  if (merchantChargeId && ability.can('view', 'Invoice')) {
+    const merchantCharge = await prisma.merchantCharge.findUnique({
+      where: {
+        id: merchantChargeId as string,
+      },
+      select: {
+        identifier: true,
+        merchantProductId: true,
+      },
+    })
+
+    if (merchantCharge && merchantCharge.identifier) {
+      const charge = await stripe.charges.retrieve(merchantCharge.identifier)
+
+      const merchantProductId = merchantCharge.merchantProductId
+      const merchantProduct = await prisma.merchantProduct.findUnique({
+        where: {
+          id: merchantProductId as string,
+        },
+        select: {
+          productId: true,
+        },
+      })
+      const productId = merchantProduct?.productId
+      const product = await getProduct({
+        where: {id: productId},
+      })
```

Comment:
*not this PR*: opportunity to pull some interactions out into the database SDK and the stripe SDK

Reply:
👍👍